bentoml/service_env.py

```diff
@@ -164,7 +187,27 @@ def save(self, path):
         requirements_txt_file = os.path.join(path, "requirements.txt")
 
         with open(requirements_txt_file, "wb") as f:
-            pip_content = "\n".join(self._pip_dependencies).encode("utf-8")
+            dependencies_map = {}
+            for dep in self._pip_dependencies:
+                name, version = parse_requirement_string(dep)
+                dependencies_map[name] = version
+            if self._auto_pip_dependencies:
+                if "__main__" in sys.modules:
```

Comment:
Instead of starting from the main module, it should start from the module where this `bento_servce` is defined. See the previous attempt where it passes in the `bento_service` instance when doing `bentoservice._env.save(..)`: https://github.com/bentoml/BentoML/compare/master...parano:bs-env-improvements?expand=1#diff-a6531af7171884b899814e04a018e143

Reply:
For example, here is the code that starts searching for local python modules(non-pypi packages, such as imported individual python files or directories with __init__.py), and 
https://github.com/bentoml/BentoML/blob/master/bentoml/bundler/bundler.py#L106

It's a bit hacky to get the module file this way but it works for most cases:
https://github.com/bentoml/BentoML/blob/master/bentoml/bundler/py_module_utils.py#L98